src/object_store/src/object/opendal_engine/opendal_object_store.rs

```diff
@@ -258,16 +266,8 @@ impl Stream for OpenDalObjectIter {
                                 let key = object.path().to_string();
                                 // FIXME: How does opendal metadata cache work?
                                 // Will below line result in one IO per object?
-                                let om = match op
-                                    .metadata(
-                                        &object,
-                                        Metakey::LastModified | Metakey::ContentLength,
-                                    )
-                                    .await
-                                {
-                                    Ok(om) => om,
-                                    Err(e) => return (Err(e), op),
-                                };
+                                let om = object.metadata();
```

Comment:
We don't need `OpenDalObjectIter` anymore, since opendal's Lister now has `metakey(Metakey::ContentLength | Metakey::LastModified)` which did the same thing here and much faster since we can reduce the `stat` call when underlying storage returns given metadata during list.

Reply:
Cool~ Thanks for the efforts!
I will remove the `OpenDalObjectIter` later.